Update Request: cryptpad 2024.12.0 → 2025.3.0

### Nixpkgs version

Please select a version.

### Package name

cryptpad

### Upstream version

2025.3.0

### Nixpkgs version

unstable

### Changelog

https://github.com/cryptpad/cryptpad/releases/tag/2025.3.0

### Additional context

_No response_

### Notify maintainers

@martinetd

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this package update does not yet exist in an [open pull request](https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+is%3Apr+label%3A%228.has%3A+package+%28update%29%22) or in [Nixpkgs Unstable](https://search.nixos.org/packages?channel=unstable).
- [x] I assert that this is not a [duplicate of any known issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%229.needs%3A+package+%28update%29%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

A naive update (updating version/sha and removing merged PR) doesn't work:
```diff
diff --git a/pkgs/by-name/cr/cryptpad/package.nix b/pkgs/by-name/cr/cryptpad/package.nix
index 94087f88b86b..0b2b5740bc4b 100644
--- a/pkgs/by-name/cr/cryptpad/package.nix
+++ b/pkgs/by-name/cr/cryptpad/package.nix
@@ -13,7 +13,7 @@
 }:
 
 let
-  version = "2024.12.0";
+  version = "2025.3.0";
   # nix version of install-onlyoffice.sh
   # a later version could rebuild from sdkjs/web-apps as per
   # https://github.com/cryptpad/onlyoffice-builds/blob/main/build.sh
@@ -83,10 +83,10 @@ buildNpmPackage {
     owner = "cryptpad";
     repo = "cryptpad";
     rev = version;
-    hash = "sha256-oSrDajaCEc7I2AsDzKoO34ffd4OeXDwFDGm45yQDSvE=";
+    hash = "sha256-NxkVMsfLzdzifdn+f0C6mBJGd1oLwcMTAIXv+gBG7rI=";
   };
 
-  npmDepsHash = "sha256-1EwxAe+8FOrngZx5+FEeu9uHKWZNBpsECEGrsyiZ2GU=";
+  npmDepsHash = "sha256-GWkyRlizPSA72WwoY+mRLwaMeD/SXdo6oUVwsd2gp7c=";
 
   nativeBuildInputs = [
     makeBinaryWrapper
@@ -99,12 +99,6 @@ buildNpmPackage {
     # fix httpSafePort setting
     # https://github.com/cryptpad/cryptpad/pull/1571
     ./0001-env.js-fix-httpSafePort-handling.patch
-    # https://github.com/cryptpad/cryptpad/pull/1740
-    (fetchpatch {
-      name = "Add `--check`, `--rdfind`, `--no-rdfind` options to `install-onlyoffice.sh`";
-      url = "https://github.com/cryptpad/cryptpad/commit/f38668735e777895db2eadd3413cff386fb12c0c.patch";
-      hash = "sha256-J4AK1XIa3q+/lD74p2c9O7jt0VEtofTmfAaQNU71sp8=";
-    })
   ];
 
   # cryptpad build tries to write in cache dir
```
```
vm-test-run-cryptpad> machine # [    9.841198] cryptpad[1014]: node:internal/event_target:1101
vm-test-run-cryptpad> machine # [    9.841599] cryptpad[1014]:   process.nextTick(() => { throw err; });
vm-test-run-cryptpad> machine # [    9.842289] cryptpad[1014]:                            ^
vm-test-run-cryptpad> machine # [    9.842660] cryptpad[1014]: Error: Cannot find module '/var/lib/cryptpad/scripts/migrations/migrate-blob-proofs.js'
vm-test-run-cryptpad> machine # [    9.844096] cryptpad[1014]:     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
vm-test-run-cryptpad> machine # [    9.844633] cryptpad[1014]:     at Function._load (node:internal/modules/cjs/loader:1055:27)
vm-test-run-cryptpad> machine # [    9.845136] cryptpad[1014]:     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
vm-test-run-cryptpad> machine # [    9.845613] cryptpad[1014]:     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
vm-test-run-cryptpad> machine # [    9.846123] cryptpad[1014]:     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:170:5)
vm-test-run-cryptpad> machine # [    9.846757] cryptpad[1014]:     at MessagePort.<anonymous> (node:internal/main/worker_thread:212:26)
vm-test-run-cryptpad> machine # [    9.847288] cryptpad[1014]:     at [nodejs.internal.kHybridDispatch] (node:internal/event_target:827:20)
vm-test-run-cryptpad> machine # [    9.847816] cryptpad[1014]:     at MessagePort.<anonymous> (node:internal/per_context/messageport:23:28) {
vm-test-run-cryptpad> machine # [    9.848356] cryptpad[1014]:   code: 'MODULE_NOT_FOUND',
vm-test-run-cryptpad> machine # [    9.848683] cryptpad[1014]:   requireStack: []
vm-test-run-cryptpad> machine # [    9.849012] cryptpad[1014]: }
vm-test-run-cryptpad> machine # [    9.849231] cryptpad[1014]: Node.js v22.14.0
```
will need to check why, will probably be a few days... Help welcome if you care about this update, I'm not really using my instance right now.

--- a/pkgs/by-name/cr/cryptpad/package.nix
+++ b/pkgs/by-name/cr/cryptpad/package.nix
@@ -14,7 +14,7 @@
 }:
 
 let
-  version = "2024.12.0";
+  version = "2025.3.0";
   # nix version of install-onlyoffice.sh
   # a later version could rebuild from sdkjs/web-apps as per
   # https://github.com/cryptpad/onlyoffice-builds/blob/main/build.sh
@@ -84,10 +84,10 @@ buildNpmPackage {
     owner = "cryptpad";
     repo = "cryptpad";
     rev = version;
-    hash = "sha256-oSrDajaCEc7I2AsDzKoO34ffd4OeXDwFDGm45yQDSvE=";
+    hash = "sha256-NxkVMsfLzdzifdn+f0C6mBJGd1oLwcMTAIXv+gBG7rI=";
   };
 
-  npmDepsHash = "sha256-1EwxAe+8FOrngZx5+FEeu9uHKWZNBpsECEGrsyiZ2GU=";
+  npmDepsHash = "sha256-GWkyRlizPSA72WwoY+mRLwaMeD/SXdo6oUVwsd2gp7c=";
 
   nativeBuildInputs = [
     makeBinaryWrapper
@@ -100,12 +100,6 @@ buildNpmPackage {
     # fix httpSafePort setting
     # https://github.com/cryptpad/cryptpad/pull/1571
     ./0001-env.js-fix-httpSafePort-handling.patch
-    # https://github.com/cryptpad/cryptpad/pull/1740
-    (fetchpatch {
-      name = "Add `--check`, `--rdfind`, `--no-rdfind` options to `install-onlyoffice.sh`";
-      url = "https://github.com/cryptpad/cryptpad/commit/f38668735e777895db2eadd3413cff386fb12c0c.patch";
-      hash = "sha256-J4AK1XIa3q+/lD74p2c9O7jt0VEtofTmfAaQNU71sp8=";
-    })
   ];
 
   # cryptpad build tries to write in cache dir
@@ -148,7 +142,7 @@ buildNpmPackage {
     # directory.
     makeWrapper "${lib.getExe nodejs}" "$out/bin/cryptpad" \
       --add-flags "$out_cryptpad/server.js" \
-      --run "for d in customize.dist lib www; do ${coreutils}/bin/ln -sf \"$out_cryptpad/\$d\" .; done" \
+      --run "for d in customize.dist lib www scripts; do ${coreutils}/bin/ln -sf \"$out_cryptpad/\$d\" .; done" \
       --run "if ! [ -d customize ]; then \"${lib.getExe nodejs}\" \"$out_cryptpad/scripts/build.js\"; fi"
   '';
 

--- a/pkgs/by-name/cr/cryptpad/package.nix
+++ b/pkgs/by-name/cr/cryptpad/package.nix
@@ -1,6 +1,7 @@
 {
   bash,
   buildNpmPackage,
+  coreutils,
   fetchFromGitHub,
   fetchpatch,
   fetchurl,
@@ -147,7 +148,7 @@ buildNpmPackage {
     # directory.
     makeWrapper "${lib.getExe nodejs}" "$out/bin/cryptpad" \
       --add-flags "$out_cryptpad/server.js" \
-      --run "for d in customize.dist lib www; do ln -sf \"$out_cryptpad/\$d\" .; done" \
+      --run "for d in customize.dist lib www; do ${coreutils}/bin/ln -sf \"$out_cryptpad/\$d\" .; done" \
       --run "if ! [ -d customize ]; then \"${lib.getExe nodejs}\" \"$out_cryptpad/scripts/build.js\"; fi"
   '';
 

